go/arrow/array/map.go

```diff
@@ -234,6 +234,10 @@ func (b *MapBuilder) AppendNulls(n int) {
 	}
 }
 
+func (b *MapBuilder) SetNull(i int) {
+	b.listBuilder.SetNull(i)
+}
```

Comment:
since this needed the extra case, can we add a test for the MapBuilder that calls `SetNull` and then `NewArray` and verify everything is correct by checking `IsNull` on the resulting array?